modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFolderServiceTest.java

```diff
@@ -91,6 +93,184 @@ public void setUp() throws Exception {
+
+		_dlAppService.addFileEntry(
+			null, _group.getGroupId(), _parentFolder.getFolderId(),
+			StringUtil.randomString() + ".jpg", ContentTypes.IMAGE_JPEG,
+			"title1", StringUtil.randomString(), StringPool.BLANK, (byte[])null,
+			null, null, serviceContext);
+
+		_dlAppService.addFileEntry(
+			null, _group.getGroupId(), _parentFolder.getFolderId(),
+			StringUtil.randomString(), ContentTypes.APPLICATION_OCTET_STREAM,
+			"title2", StringUtil.randomString(), StringPool.BLANK, (byte[])null,
+			null, null, serviceContext);
+
+		Assert.assertEquals(
+			0,
+			_dlFolderService.getFoldersAndFileEntriesAndFileShortcutsCount(
+				_alternativeGroup.getGroupId(), _parentFolder.getFolderId(),
+				WorkflowConstants.STATUS_APPROVED,
+				ArrayUtil.toStringArray(DLUtil.getAllMediaGalleryMimeTypes()),
+				false));
+	}
+
+	@Test
+	public void testShouldReturnOnlyAcceptedMimeTypes() throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId());
+
+		serviceContext.setAttribute(
+			"fileEntryTypeId", _dlFileEntryType.getFileEntryTypeId());
+
+		List<FileEntry> expectedFileEntries = new ArrayList<>();
+
+		_dlAppService.addFileEntry(
+			null, _group.getGroupId(), _parentFolder.getFolderId(),
+			StringUtil.randomString(), ContentTypes.APPLICATION_OCTET_STREAM,
+			"title2", StringUtil.randomString(), StringPool.BLANK, (byte[])null,
+			null, null, serviceContext);
+
+		_dlAppService.addFileEntry(
+			null, _group.getGroupId(), _parentFolder.getFolderId(),
+			StringUtil.randomString(), ContentTypes.APPLICATION_OCTET_STREAM,
+			"title1", StringUtil.randomString(), StringPool.BLANK, (byte[])null,
+			null, null, serviceContext);
+
+		FileEntry fileEntry = _dlAppService.addFileEntry(
+			null, _group.getGroupId(), _parentFolder.getFolderId(),
+			StringUtil.randomString() + ".jpg", ContentTypes.IMAGE_JPEG,
+			"title3", StringUtil.randomString(), StringPool.BLANK, (byte[])null,
+			null, null, serviceContext);
+
+		expectedFileEntries.add(fileEntry);
+
+		List<Object> actualFoldersAndFileEntriesAndFileShortcuts =
+			_dlFolderService.getFoldersAndFileEntriesAndFileShortcuts(
+				_group.getGroupId(), _parentFolder.getFolderId(),
+				ArrayUtil.toStringArray(DLUtil.getAllMediaGalleryMimeTypes()),
+				_dlFileEntryType.getFileEntryTypeId(), false,
+				WorkflowConstants.STATUS_APPROVED, QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS,
+				new RepositoryModelTitleComparator<FileEntry>(true));
+
+		Assert.assertEquals(
+			actualFoldersAndFileEntriesAndFileShortcuts.toString(), 1,
+			actualFoldersAndFileEntriesAndFileShortcuts.size());
+
+		for (int i = 0; i < actualFoldersAndFileEntriesAndFileShortcuts.size();
+			 i++) {
+
+			FileEntry expectedFileEntry = expectedFileEntries.get(i);
+			DLFileEntry actualFileEntry =
+				(DLFileEntry)actualFoldersAndFileEntriesAndFileShortcuts.get(i);
+
+			Assert.assertEquals(
+				expectedFileEntry.getFileEntryId(),
+				actualFileEntry.getFileEntryId());
+		}
```

Comment:
Here you already know the list size is 1, so why not get rid of the for loop? Also, you must use a different suffix FileEntry and DLFileEntry variables (e.g. actualDLFileEntry vs actualFileEntry).

Reply:
Thanks, I will add a commit renaming the variables of the file.